src/watcher/appointmentSubscriber.ts

```diff
@@ -0,0 +1,96 @@
+import { ethers } from "ethers";
+import { ApplicationError } from "../dataEntities/errors";
+
+/** A helper interface for the additional markup we require on a listener */
+interface IAppointmentListener {
+    (...args: any[]): void;
+    appointmentId: string;
+}
+
+/**
+ * If we wish to unsubscribe an appointment from the provider we have a problem. In the provider
+ * appointments are keyed only by their filter, so an appointment could be updated, then another
+ * process tries to remove the old appointment and in doing so removes the new one. This class
+ * assigns appointment ids to listeners and ensures that appointments are only removed from the
+ * provided if the provide appointment id matches.
```

Comment:
Typo in the last sentence of the comment.

Reply:
fixed